tests/account_change_password_test.go

```diff
@@ -100,6 +101,83 @@ func TestChangePassword(t *testing.T) {
 			StatusCode: 200,
 		})
 	})
+
+	// sytest: Pushers created with a different access token are deleted on password change
+	t.Run("Pushers created with a different access token are deleted on password change", func(t *testing.T) {
+		sessionOptional := createSession(t, deployment, "test_change_password_user", "new_optional_password")
+		reqBody := client.WithJSONBody(t, map[string]interface{}{
+			"data": map[string]interface{}{
+				"url": "https://dummy.url/_matrix/push/v1/notify",
+			},
+			"profile_tag":         "tag",
+			"kind":                "http",
+			"app_id":              "complement",
+			"app_display_name":    "complement_display_name",
+			"device_display_name": "device_display_name",
+			"pushkey":             "a_push_key",
+			"lang":                "en",
+		})
+
+		res := sessionOptional.DoFunc(t, "POST", []string{"_matrix", "client", "r0", "pushers", "set"}, reqBody)
+
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 200,
+		})
+
+		changePassword(t, passwordClient, "new_optional_password", newPassword)
```

Comment:
I would use `password1` `password2` and `password3` instead of `"new_optional_password"` and `newPassword`, which becomes confusing when `newPassword` is used as an old password arg!